File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        if n == 1:
            return
        i = 0
        j = 1
        for j in range(n):
            if nums[j] != 0:
                nums[i], nums[j] = nums[j] , nums[i]
                i += 1
            

        
        

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i =0;
        int j = nums.length-1;

        if(j==0)
            return;

        while(i<j){
            if(nums[i] != 0)
                i++;
            else{
                for(int k=i; k<j; k++){
                    int temp = nums[k];
                    nums[k] = nums[k+1];
                    nums[k+1] = temp;
                }
                j--;
            }
        }
    }
}